src/lib.rs

```diff
@@ -495,7 +496,7 @@ pub(crate) fn create_emitter<'a>(config: &Config) -> Box<dyn Emitter + 'a> {
 impl<'b, T: Write + 'b> Drop for Session<'b, T> {
     fn drop(&mut self) {
         if let Some(ref mut out) = self.out {
-            let _ = self.emitter.emit_footer(out);
+            let _ = self.emitter.emit_footer(out, self.errors.has_diff);
```

Comment:
This was the best approach I could see to know whether there were any files with diffs when the final output is emitted in `emit_footer`. The only alternatives I could see involved trying to maintain state about diffed files in the emitters which felt wrong.